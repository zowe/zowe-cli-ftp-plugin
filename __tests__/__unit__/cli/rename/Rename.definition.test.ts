/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */

import * as RenameDefinition from "../../../../src/cli/rename/Rename.definition";

describe("Rename definition", () => {

    it("should match the snapshot", () => {
        RenameDefinition.children.forEach((child) => {
            child.handler = child.handler.replace(/^.*[\\/]/, "");
        });
        expect(RenameDefinition).toMatchSnapshot();
    });

});
