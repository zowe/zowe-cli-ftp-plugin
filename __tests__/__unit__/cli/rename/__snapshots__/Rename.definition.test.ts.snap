// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rename definition should match the snapshot 1`] = `
Object {
  "aliases": Array [
    "mv",
  ],
  "children": Array [
    Object {
      "aliases": Array [
        "ds",
      ],
      "description": "Rename a cataloged data set",
      "examples": Array [
        Object {
          "description": "Rename the data set ibmuser.jcl to ibmuser.cntl",
          "options": "ibmuser.jcl ibmuser.cntl",
        },
        Object {
          "description": "Rename the data set member \\"ibmuser.cntl(alloc)\\" to \\"ibmuser.cntl(alloc2)\\". Note: you can only rename members within the same partitioned data set. You cannot move a member to another data set with this command.",
          "options": "\\"ibmuser.cntl(alloc)\\" \\"ibmuser.cntl(alloc2)\\"",
        },
      ],
      "handler": "DataSet.Handler",
      "name": "data-set",
      "positionals": Array [
        Object {
          "description": "The current name of the data set you want to rename.",
          "name": "oldDataSet",
          "required": true,
          "type": "string",
        },
        Object {
          "description": "The new name for the data set.",
          "name": "newDataSet",
          "required": true,
          "type": "string",
        },
      ],
      "profile": Object {
        "optional": Array [
          "zftp",
        ],
      },
      "summary": "Rename a cataloged data set",
      "type": "command",
    },
    Object {
      "aliases": Array [
        "uss",
        "uf",
      ],
      "description": "Rename a USS file or directory",
      "examples": Array [
        Object {
          "description": "Rename the file /u/users/ibmuser/hello.txt to /u/users/ibmuser/hello2.txt",
          "options": "\\"/u/users/ibmuser/hello.txt\\" \\"/u/users/ibmuser/hello2.txt\\"",
        },
      ],
      "handler": "UssFile.Handler",
      "name": "uss-file",
      "positionals": Array [
        Object {
          "description": "The current name of the USS file you want to rename.",
          "name": "olduss",
          "required": true,
          "type": "string",
        },
        Object {
          "description": "The new name for the USS file.",
          "name": "newuss",
          "required": true,
          "type": "string",
        },
      ],
      "profile": Object {
        "optional": Array [
          "zftp",
        ],
      },
      "summary": "Rename a USS file or directory",
      "type": "command",
    },
  ],
  "description": "Rename data sets and USS files or directories",
  "name": "rename",
  "passOn": Array [
    Object {
      "ignoreNodes": Array [
        Object {
          "type": "group",
        },
      ],
      "merge": true,
      "property": "options",
      "value": Array [
        Object {
          "aliases": Array [
            "H",
          ],
          "description": "The hostname or IP address of the z/OS server to connect to.",
          "group": "FTP Connection options",
          "name": "host",
          "type": "string",
        },
        Object {
          "aliases": Array [
            "P",
          ],
          "defaultValue": 21,
          "description": "The port of the z/OS FTP server.",
          "group": "FTP Connection options",
          "name": "port",
          "type": "number",
        },
        Object {
          "aliases": Array [
            "u",
          ],
          "description": "Username for authentication on z/OS",
          "group": "FTP Connection options",
          "name": "user",
          "type": "string",
        },
        Object {
          "aliases": Array [
            "p",
            "pass",
            "pw",
          ],
          "description": "Password to authenticate to FTP.",
          "group": "FTP Connection options",
          "name": "password",
          "type": "string",
        },
        Object {
          "defaultValue": true,
          "description": "Set to true for both control and data connection encryption, 'control' for control connection encryption only, or 'implicit' for implicitly encrypted control connection (this mode is deprecated in modern times, but usually uses port 990). Note: Unfortunately, this plugin's functionality only works with FTP and FTPS, not 'SFTP' which is FTP over SSH.",
          "group": "FTP Connection options",
          "name": "secure-ftp",
          "type": "boolean",
        },
        Object {
          "aliases": Array [
            "ru",
          ],
          "defaultValue": null,
          "description": "Reject self-signed certificates. Only specify this if you are connecting to a secure FTP instance.",
          "group": "TLS / Secure Connection options",
          "name": "reject-unauthorized",
          "type": "boolean",
        },
        Object {
          "aliases": Array [
            "sn",
          ],
          "defaultValue": null,
          "description": "Server name for the SNI (Server Name Indication) TLS extension. Only specify if you are connecting securely",
          "group": "TLS / Secure Connection options",
          "name": "server-name",
          "type": "string",
        },
        Object {
          "aliases": Array [
            "ct",
          ],
          "defaultValue": 10000,
          "description": "How long (in milliseconds) to wait for the control connection to be established.",
          "group": "FTP Connection options",
          "name": "connection-timeout",
          "type": "number",
        },
      ],
    },
  ],
  "summary": "Rename data sets and USS files or directories",
  "type": "group",
}
`;
